src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/ReturnTaglet.java

```diff
@@ -54,47 +54,77 @@
 public class ReturnTaglet extends BaseTaglet implements InheritableTaglet {
 
     public ReturnTaglet() {
-        super(DocTree.Kind.RETURN, false, EnumSet.of(Location.METHOD));
+        super(DocTree.Kind.RETURN, true, EnumSet.of(Location.METHOD));
+    }
+
+    @Override
+    public boolean isBlockTag() {
+        return true;
     }
 
     @Override
     public void inherit(DocFinder.Input input, DocFinder.Output output) {
-        List<? extends ReturnTree> tags = input.utils.getReturnTrees(input.element);
-        CommentHelper ch = input.utils.getCommentHelper(input.element);
+        Utils utils = input.utils;
+        CommentHelper ch = utils.getCommentHelper(input.element);
+
+        ReturnTree tag = null;
+        List<? extends ReturnTree> tags = utils.getReturnTrees(input.element);
         if (!tags.isEmpty()) {
+            tag = tags.get(0);
+        } else {
+            List<? extends DocTree> firstSentence = utils.getFirstSentenceTrees(input.element);
+            if (firstSentence.size() == 1 && firstSentence.get(0).getKind() == DocTree.Kind.RETURN) {
```

Comment:
There's one more place where `firstSentence.size() == 1` is checked. I wonder if it could be a simpler `!firstSentence.isEmpty()` check?

Are there any non-trivial cases which might bite us later, should we change that check?

Reply:
I think the code is better as is.  It would be wrong to have a firstSentence containing multiple items, beginning with `{@return}` and then to only return the first one in the following line of code. Arguably, a better form may be to use `isEmpty()` and then subsequently assert (somehow) that the size is exactly 1, but using assertions like that is a different discussion (as you know!)

Reply:
I guess I was thinking about a corner case like `{@summary {@return ...} ...}` which this patch and the complementary specification consider illegal anyway.